### DISCUSSION TITLE
[BUG] - Task not being called

### DISCUSSION THREAD

**Discussion by okridgway** [go-co-op/gocron#266] [Q&A] [answered]
### Describe the bug

I've written a simple program but can't for the life of me figure out why it isn't executing the task function.
### To Reproduce
First I create a new scheduler:
```go
	s := gocron.NewScheduler(time.UTC)
```
Then I specify how often the task should occur (every 10 seconds):
```go
	s.Every(10).Do(task)
```
Finally, I start the task with:
```go
	s.StartAsync()
```
But the task function (previously defined to print "Task is being performed" never executes. A playground link can be found [here](https://go.dev/play/p/eC1vYLblxdI).

### Version

I am using version `v1.10.0`.
### Expected behavior
I expected the program to call the task function.
### Additional context

I've tried adding a `select{}` statement at the end of my code as suggested [here](https://stackoverflow.com/questions/43314088/cron-job-in-go-not-running-as-expected), but all to no avail.

**Comment by JohnRoesler** [comment]
Hi @okridgway 

Running the playground link you provided I see it's going to always evaluate true as the scheduler `s` is not nil and it logs fatal outputting the memory addresses of the scheduler struct. I also don't see that the scheduler is being started.

Try this:

```golang
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/go-co-op/gocron"
)

var task = func() {
	fmt.Println("Task is being performed.")
}

func main() {
	fmt.Println("Task is starting.")

	s := gocron.NewScheduler(time.UTC)
	_, err := s.Every(10).Do(task)
	if err != nil {
		log.Fatal(s)
	}

	s.StartBlocking()
}
```

It timed out for me in the playground, but runs locally just fine.

**Comment by okridgway** [comment]
Ok, I've got it working now (thank you!) but don't quite understand why `s.Every(10).Do(task)` has to be set to a variable in order for it to work, as the primary example on the main README suggests that it doesn't need to be set to a variable. Have things just changed? Or am I missing something else?

**Comment by JohnRoesler** [comment, marked as answer]
Can you paste the code you're running where you're seeing that have to set it to a variable? The README is correct.

For example, this works for me:

```golang
package main

import (
	"fmt"
	"time"

	"github.com/go-co-op/gocron"
)

var task = func() {
	fmt.Println("Task is being performed.")
}

func main() {
	fmt.Println("Task is starting.")

	s := gocron.NewScheduler(time.UTC)
	s.Every(10).Do(task)

	s.StartBlocking()
}
```

**Comment by JohnRoesler** [comment]
@okridgway I transferred your issue here as I deem it a discussion vs. a bug with gocron - I'm happy to continue helping here!

**Comment by Pr3c10us** [comment]
The job doesn't execute the task, i have tried it without the AT() and it works fine, i.e _, err := newScheduler.Every(1).Second().Do(cron.Task("jobDetails.UserSlackId", "jobDetails.WorkspaceSlackId", uuid.New()))
	if err != nil {
		panic(err)
	}
	
	code:
	`
job := scheduler.Name(jobDetails.Id.String()).Every(jobDetails.Intervals).Weeks()
	for _, day := range weekDays {
		if day == "Monday" {
			job = job.Monday()
		} else if day == "Tuesday" {
			job = job.Tuesday()
		} else if day == "Wednesday" {
			job = job.Wednesday()
		} else if day == "Thursday" {
			job = job.Thursday()
		} else if day == "Friday" {
			job = job.Friday()
		} else if day == "Saturday" {
			job = job.Saturday()
		} else if day == "Sunday" {
			job = job.Sunday()
		} else {
			return appError.BadRequest(errors.New("invalid day of the week"))
		}
	}
	_, err := job.At(jobDetails.Time).Tag(jobDetails.Id.String()).Do(task, jobDetails.UserSlackId, jobDetails.WorkspaceSlackId, jobDetails.WorkflowId)
	if err != nil {
		return errors.New("error scheduling the task")
	}
`
